pkgs/development/compilers/ballerina/default.nix

```diff
@@ -0,0 +1,48 @@
+{ ballerina, lib, writeText, runCommand, makeWrapper, fetchzip, stdenv, jdk11 }:
+let
```

Comment:
What shall we do about `codeName`? Since we need a let block for `codeName` anyway, I figured it would be more convenient for future updates to also put the SHA256 and the version there, seeing that these will all change for the same reason in the future. WDYT?

Reply:
Won’t help us, since the URL contains both version in isolation and version+codename. Is there anything particularly prohibitive about binding variables? We could hard-code the code-name in the URL, but since this is likely to change in conjunction with the version in the future, it seems preferable to name them explicitly.

Reply:
Shall we resolve this one?

Reply:
LGTM